projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -2,33 +2,77 @@ import { getRedirectUrl } from '@automattic/jetpack-components';
 import { getSiteFragment, useAnalytics } from '@automattic/jetpack-shared-extension-utils';
 import { Modal, Button, CheckboxControl } from '@wordpress/components';
 import { usePrevious } from '@wordpress/compose';
-import { useSelect } from '@wordpress/data';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
-import { useEffect, useState } from '@wordpress/element';
+import { useEffect, useRef, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import './editor.scss';
 
 export const name = 'launchpad-save-modal';
 
 export const settings = {
+	// TODO: Rename LaunchpadSaveModal to something more generic
 	render: function LaunchpadSaveModal() {
-		const isSavingSite = useSelect(
-			select => select( editorStore ).isSavingNonPostEntityChanges(),
-			[]
-		);
-		const isSavingPost = useSelect( select => select( editorStore ).isSavingPost(), [] );
+		// First Post Published Modal
+		const postType = useSelect( select => select( editorStore ).getCurrentPostType(), [] );
+		const { link } = useSelect( select => select( 'core/editor' ).getCurrentPost() );
 		const isCurrentPostPublished = useSelect(
 			select => select( editorStore ).isCurrentPostPublished(),
 			[]
 		);
+		const shouldShowFirstPostPublishedModal = useSelect( select =>
+			select( 'automattic/wpcom-welcome-guide' ).getShouldShowFirstPostPublishedModal()
+		);
+		const {
+			fetchShouldShowFirstPostPublishedModal,
+			setShouldShowFirstPostPublishedModal,
+		} = useDispatch( 'automattic/wpcom-welcome-guide' );
+		const prevIsCurrentPostPublished = useRef( isCurrentPostPublished );
+		const [ displayFirstPostPublishedModal, setDisplayFirstPostPublishedModal ] = useState( false );
+
+		useEffect( () => {
+			fetchShouldShowFirstPostPublishedModal();
+		}, [ fetchShouldShowFirstPostPublishedModal ] );
+
+		useEffect( () => {
+			// If the user is set to see the first post modal and current post status changes to publish,
+			// open the post publish modal
+			if (
+				shouldShowFirstPostPublishedModal &&
+				! prevIsCurrentPostPublished.current &&
+				isCurrentPostPublished &&
+				postType === 'post'
+			) {
+				prevIsCurrentPostPublished.current = isCurrentPostPublished;
+				setShouldShowFirstPostPublishedModal( false );
+
+				// When the post published panel shows, it is focused automatically.
+				// Thus, we need to delay open the modal so that the modal would not be close immediately
+				// because the outside of modal is focused
+				window.setTimeout( () => {
+					setDisplayFirstPostPublishedModal( true );
+				} );
+			}
+		}, [
+			postType,
+			shouldShowFirstPostPublishedModal,
+			isCurrentPostPublished,
+			setShouldShowFirstPostPublishedModal,
+		] );
 
+		// Launchpad Modal
```

Comment:
**NOTE:** The original launchpad modal logic remains relatively untouched except for some naming changes. This is meant to accommodate the existence of two different modals in this file now ( launchpad modal vs. first post published modal )